sqld/src/hrana/mod.rs

```diff
@@ -45,3 +52,20 @@ pub async fn serve(db_factory: Arc<dyn DbFactory>, bind_addr: SocketAddr) -> Res
         }
     }
 }
+
+pub fn load_jwt_key(path: &Path) -> Result<jsonwebtoken::DecodingKey> {
+    let data = fs::read(path)?;
+    if data.starts_with(b"-----BEGIN PUBLIC KEY-----") {
+        jsonwebtoken::DecodingKey::from_ed_pem(&data)
+            .context("Could not decode Ed25519 public key from PEM")
+    } else if data.starts_with(b"-----BEGIN PRIVATE KEY-----") {
+        bail!("Received a private key, but a public key is expected")
+    } else if data.starts_with(b"-----BEGIN") {
+        bail!("Key is in unsupported PEM format")
+    } else if let Ok(data_str) = str::from_utf8(&data) {
+        jsonwebtoken::DecodingKey::from_ed_components(&data_str)
+            .context("Could not decode Ed25519 public key from base64")
+    } else {
+        bail!("Key is in an unsupported binary format")
+    }
+}
```

Comment:
This code would certainly be useful in auth!

Reply:
The code in `http/auth.rs` seems to implement a diferent mechanism, instead of passing an argument pointing to a file with Ed25519 pubkey, we pass an argument that directly contains a base64-encoded RSA pubkey in PKCS#8 (which is awkwardly decoded by wrapping it in PEM headers and parsing it as PEM). However, the JWT auth in HTTP is not yet used in Turso, so perhaps we can use the Hrana mechanism also in HTTP?

Reply:
See clerk: https://clerk.dev/docs/request-authentication/validate-session-tokens?utm_source=www.google.fr&utm_medium=referral&utm_campaign=none

Reply:
We do a similar thing with TLS certificates and keys for gRPC: they are passed as env variables to the container and the entrypoint script stores them in temporary files and passes the paths to these files to sqld via the command line arguments.

Now that I think about it again, this is rather silly: it would be better if all keys/certs were passed to sqld in one of two ways:

- With a command line argument pointing to a file (useful when you run sqld standalone)
- With an env variable containing the data (useful when you run sqld in a container)

However, I'm not sure how we could implement this scheme with clap?

Reply:
Maybe @CodingDoug and @athoscouto input would be useful here

Reply:
I added the option of passing the key directly in env SQLD_HRANA_JWT_KEY, so we won't need to create temporary files in a container.